### ISSUE TITLE
Issue with newick tree parser

### ISSUE THREAD

**Issue by shz9** [biopython/biopython#1782]
Hi all,

I'm having some issues with the newick tree parser in Biopython. In particular, I think the parser doesn't work properly with the tree format that you obtain from PAML.

### Setup 

I'm using python version 2.7.5 on Linux (CentOS) with Biopython v. 1.71. I'm trying to parse and transform the following tree structure that I obtained from PAML 4.9:

```
((((Equus_caballus: 0.101350, Sus_scrofa: 0.162050): 0.006402, (Pteropus_vampyrus: 0.170581, Carollia_perspicillata: 0.146190): 0.018652): 0.031467, (Pan_troglodytes: 0.005090, Homo_sapiens: 0.006653): 0.147803): 0.013442, (Dasypus_novemcinctus: 0.176523, ((Trichechus_manatus: 0.056184, Loxodonta_africana: 0.073775): 0.022719, (Chrysochloris_asiatica: 0.203039, Orycteropus_afer: 0.121019): 0.015696): 0.182032): 0.017545);
```

### Steps to reproduce

```
>>> from Bio import Phylo
>>> tree_obj = Phylo.read("/path/to/tree/file.nwk", "newick")
>>> list(tree_obj.find_clades())
[Clade(), Clade(confidence=0.013442), Clade(confidence=0.031467), Clade(confidence=0.006402), Clade(name='0.101350'), Clade(name='0.162050'), Clade(confidence=0.018652), Clade(name='0.170581'), Clade(name='0.146190'), Clade(confidence=0.147803), Clade(name='0.005090'), Clade(name='0.006653'), Clade(confidence=0.017545), Clade(name='0.176523'), Clade(confidence=0.182032), Clade(confidence=0.022719), Clade(name='0.056184'), Clade(name='0.073775'), Clade(confidence=0.015696), Clade(name='0.203039'), Clade(name='0.121019')]
```

### Expected behavior

As you can see in the tree structure above, I don't have any confidence values. Only species names and branch lengths. I'm expecting the tree parser to populate the name and branch_length attributes only, but that's not what I'm observing.

### Is it an issue with non-standard Newick format?

At first, I thought it might be an issue with non-standard newick format. I realize that there are some variations on the format and some parsers don't work with all variations, but I'm no expert. I tried parsing the tree structure on some other platforms and they produced the expected behavior.

For example, using the `ape` package in `R`, I was able to parse the tree structure just fine:

```
> library(ape)
> x <- read.tree(text = "((((Equus_caballus: 0.101350, Sus_scrofa: 0.162050): 0.006402, (Pteropus_vampyrus: 0.170581, Carollia_perspicillata: 0.146190): 0.018652): 0.031467, (Pan_troglodytes: 0.005090, Homo_sapiens: 0.006653): 0.147803): 0.013442, (Dasypus_novemcinctus: 0.176523, ((Trichechus_manatus: 0.056184, Loxodonta_africana: 0.073775): 0.022719, (Chrysochloris_asiatica: 0.203039, Orycteropus_afer: 0.121019): 0.015696): 0.182032): 0.017545);")
> x
Phylogenetic tree with 11 tips and 10 internal nodes.

Tip labels:
	Equus_caballus, Sus_scrofa, Pteropus_vampyrus, Carollia_perspicillata, Pan_troglodytes, Homo_sapiens, ...

Rooted; includes branch lengths.
```

I also tried visualizing the tree structure on some of the standard web UI's out there (e.g. [ETE Toolkit](http://etetoolkit.org/treeview/), [T-REX](http://www.trex.uqam.ca/index.php?action=newick&project=trex), etc.), and all of them worked fine with the format above.

**Comment by peterjc** [comment]
Reformulation of your test case:

```
$ python3 -c "from io import StringIO; from Bio import Phylo; print(Phylo.read(StringIO('((((Equus_caballus: 0.101350, Sus_scrofa: 0.162050): 0.006402, (Pteropus_vampyrus: 0.170581, Carollia_perspicillata: 0.146190): 0.018652): 0.031467, (Pan_troglodytes: 0.005090, Homo_sapiens: 0.006653): 0.147803): 0.013442, (Dasypus_novemcinctus: 0.176523, ((Trichechus_manatus: 0.056184, Loxodonta_africana: 0.073775): 0.022719, (Chrysochloris_asiatica: 0.203039, Orycteropus_afer: 0.121019): 0.015696): 0.182032): 0.017545);'), 'newick'))"
Tree(rooted=False, weight=1.0)
    Clade()
        Clade(confidence=0.013442)
            Clade(confidence=0.031467)
                Clade(confidence=0.006402)
                    Clade(name='0.101350')
                    Clade(name='0.162050')
                Clade(confidence=0.018652)
                    Clade(name='0.170581')
                    Clade(name='0.146190')
            Clade(confidence=0.147803)
                Clade(name='0.005090')
                Clade(name='0.006653')
        Clade(confidence=0.017545)
            Clade(name='0.176523')
            Clade(confidence=0.182032)
                Clade(confidence=0.022719)
                    Clade(name='0.056184')
                    Clade(name='0.073775')
                Clade(confidence=0.015696)
                    Clade(name='0.203039')
                    Clade(name='0.121019')
```

Going instead to the Newick parser inside the Nexus module, things seem fine:

```
$ python3 -c "from Bio.Nexus.Trees import Tree; tree = Tree('((((Equus_caballus: 0.101350, Sus_scrofa: 0.162050): 0.006402, (Pteropus_vampyrus: 0.170581, Carollia_perspicillata: 0.146190): 0.018652): 0.031467, (Pan_troglodytes: 0.005090, Homo_sapiens: 0.006653): 0.147803): 0.013442, (Dasypus_novemcinctus: 0.176523, ((Trichechus_manatus: 0.056184, Loxodonta_africana: 0.073775): 0.022719, (Chrysochloris_asiatica: 0.203039, Orycteropus_afer: 0.121019): 0.015696): 0.182032): 0.017545);'); print(tree.to_string(branchlengths_only=True))"
tree a_tree = ((((Equus_caballus:0.10135,Sus_scrofa:0.16205):0.00640,(Pteropus_vampyrus:0.17058,Carollia_perspicillata:0.14619):0.01865):0.03147,(Pan_troglodytes:0.00509,Homo_sapiens:0.00665):0.14780):0.01344,(Dasypus_novemcinctus:0.17652,((Trichechus_manatus:0.05618,Loxodonta_africana:0.07377):0.02272,(Chrysochloris_asiatica:0.20304,Orycteropus_afer:0.12102):0.01570):0.18203):0.01755);
```

**Comment by etal** [comment]
Is the difference due to the extra space after `taxon_name:` ? That should be easy enough to fix.

**Comment by peterjc** [comment]
I think you are right about the white space here - giving the space-less version which I copied and pasted from the Nexus module's output, this works:

```
$ python3 -c "from io import StringIO; from Bio import Phylo; print(Phylo.read(StringIO('((((Equus_caballus:0.10135,Sus_scrofa:0.16205):0.00640,(Pteropus_vampyrus:0.17058,Carollia_perspicillata:0.14619):0.01865):0.03147,(Pan_troglodytes:0.00509,Homo_sapiens:0.00665):0.14780):0.01344,(Dasypus_novemcinctus:0.17652,((Trichechus_manatus:0.05618,Loxodonta_africana:0.07377):0.02272,(Chrysochloris_asiatica:0.20304,Orycteropus_afer:0.12102):0.01570):0.18203):0.01755);'), 'newick'))"
Tree(rooted=False, weight=1.0)
    Clade()
        Clade(branch_length=0.01344)
            Clade(branch_length=0.03147)
                Clade(branch_length=0.0064)
                    Clade(branch_length=0.10135, name='Equus_caballus')
                    Clade(branch_length=0.16205, name='Sus_scrofa')
                Clade(branch_length=0.01865)
                    Clade(branch_length=0.17058, name='Pteropus_vampyrus')
                    Clade(branch_length=0.14619, name='Carollia_perspicillata')
            Clade(branch_length=0.1478)
                Clade(branch_length=0.00509, name='Pan_troglodytes')
                Clade(branch_length=0.00665, name='Homo_sapiens')
        Clade(branch_length=0.01755)
            Clade(branch_length=0.17652, name='Dasypus_novemcinctus')
            Clade(branch_length=0.18203)
                Clade(branch_length=0.02272)
                    Clade(branch_length=0.05618, name='Trichechus_manatus')
                    Clade(branch_length=0.07377, name='Loxodonta_africana')
                Clade(branch_length=0.0157)
                    Clade(branch_length=0.20304, name='Chrysochloris_asiatica')
                    Clade(branch_length=0.12102, name='Orycteropus_afer')
```

**Comment by nimne** [comment]
It looks like the right place to strip the spaces is in `Phylo/NewickIO.py` here:
```python
    def _parse_tree(self, text):
        """Parse the text representation into an Tree object (PRIVATE)."""
        tokens = re.finditer(tokenizer, text.strip())
```
to this:
```python
    def _parse_tree(self, text):
        """Parse the text representation into an Tree object (PRIVATE)."""
        tokens = re.finditer(tokenizer, text.strip().replace(': ', ':'))
```

Does this have the potential to break anything? All I can find is that spaces are generally ignored during parsing, so this seems safe.

**Comment by etal** [comment]
That should be safe, but the extra `.replace(': ', ':')` might slow down the parser overall. I'd like to see if the regex `tokenizer` can be updated to skip over these extra spaces.

**Comment by peterjc** [comment]
Should be fixed by #1821, although we don't have a new unittest for this yet if anyone wanted to work on that? Thanks!
